### ISSUE TITLE
`docker compose` behaves differently than `docker-compose` for volume mounting

### ISSUE THREAD

**Issue by DaXdata** [docker/cli#3127]
**Description**

Volume mounting source path is affected by the location of the docker-compose.yml file.
The drive partition letter is added as a prefix before the volume monting path.

If the docker-compose.yml file is located under the C drive, the prefix "C:\\" is added to the volume source path.
The same goes if the docker-compose.yml file is located on the D drive. the prefix "D:\\" is then added to the volume source path.

**Steps to reproduce the issue:**
1. Content of D:/VolumeTest/docker-compose.yml file:
```yml
version: '3.9'

services:
  VolumeTest:
    image: debian:buster-slim
    container_name: VolumeTest
    volumes:
      - "/C/MountVolume:/etc/MountVolume"
```

2. Start container using docker-cli
```sh
docker compose up -d
```

3. Inspect mounting paht:
```sh
docker inspect VolumeTest
```

4. Stop container
```sh
docker compose down
```

5.  Start container using docker-compose
```sh
docker-compose up -d
```
6. Inspect mounting paht:
```sh
docker inspect VolumeTest
```

7. Stop container
```sh
docker-compose down
```

**Describe the results you received:**
***Docker CLI output (step 3)***
```sh
        "Mounts": [
            {
                "Type": "bind",
                "Source": "D:\\C\\MountVolume",
                "Destination": "/etc/MountVolume",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
```

***Docker-Compose output (step 6)***
```sh
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/C/MountVolume",
                "Destination": "/etc/MountVolume",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
```

**Describe the results you expected:**
The result i would expect is what docker-compose produces (step 6).

**Additional information you deem important (e.g. issue happens only occasionally):**
Consistent bug.

**Output of `docker version`:**

```sh
Docker version 20.10.6, build 370c289
```

**Output of `docker-compose version`:**
```sh
docker-compose version 1.29.1, build c34c88b2
```

**Output of `docker info`:**

```sh
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)
  compose: Docker Compose (Docker Inc., 2.0.0-beta.1)
  scan: Docker Scan (Docker Inc., v0.8.0)

Server:
 Containers: 12
  Running: 9
  Paused: 0
  Stopped: 3
 Images: 19
 Server Version: 20.10.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc version: 12644e614e25b05da6fd08a38ffa0cfe1903fdec
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.4.72-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 24.94GiB
 Name: docker-desktop
 ID: 5QJX:SADL:FTRP:TD6M:2NOT:AFSF:63XY:T5JU:OPMK:QVNV:354V:TOSD
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Windows 10 version 20H2 (19042.985)
WSL 2

**Comment by ndeloof** [comment]
I'll investigate this issue, need to understand how docker/cli do address such linux-style windows paths and use the same approach in docker/compose-cli (by the way, please report `docker compose` issue on this repository)

Any reason you use `/C/MountVolume` syntax, and not the Windows syntax `C:\MountVolume` that would feel natural for a windows user?

**Comment by ndeloof** [comment]
Hi,

this issue has already been fixed by https://github.com/docker/compose-cli/pull/1704 
Fix is available in [Compose v2.0.0-beta.3](https://github.com/docker/compose-cli/releases/tag/v2.0.0-beta.3)
